arlo_server/models.py

```diff
@@ -332,3 +332,16 @@ class File(db.Model):
     name = db.Column(db.String(250), nullable=False)
     contents = db.Column(db.Text, nullable=False)
     uploaded_at = db.Column(db.DateTime(timezone=False), nullable=False)
+
+    # Metadata for processing files in the background.
+    processing_started_at = db.Column(db.DateTime(timezone=False), nullable=True)
+    processing_succeeded_at = db.Column(db.DateTime(timezone=False), nullable=True)
+    processing_errored_at = db.Column(db.DateTime(timezone=False), nullable=True)
```

Comment:
What happens with these fields if it errors on the first attempt, then succeeds on the second attempt? (I'm assuming you can retry, which maybe is not the case.) Would it be simpler to have one field `processing_ended_at` and rely on `processing_error` as the flag for whether or not it ended in success or error?

Reply:
That's more or less what @benadida suggested, and I just implemented that. Right now there's no way to retry processing, you just upload a new file.